src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2442,5 +2445,163 @@ private void InterpretLoadElement(int token)
                     throw new NotImplementedException();
             }
         }
+
+        // Holds information about a method call
+        private struct MethodCallInfo
+        {
+            public int ArgCount;
+            public IntPtr MethodAddress;
+            public IntPtr UnboxingStubAddress;
+            public CallConverter.CallingConvention CallingConvention;
+            public TypeLoaderEnvironment.MethodAddressType MethodAddressType;
+            public LocalVariableType[] LocalVariableTypes;
+            public TypeDesc ReturnType;
+            public StackItem ReturnValue;
+        }
+
+        private void InterpretCallDelegate(ref MethodCallInfo callInfo, ref LocalVariableSet localVariableSet)
+        {
+            for (int i = callInfo.ArgCount; i > 0; i--)
+            {
+                StackItem stackItem = PopWithValidation();
+                switch (stackItem.Kind)
```

Comment:
I think this will need to use the types of the method arguments, not types of the stack slot.

E.g. if the method takes a `float` argument, we should not write a `double` here. Can you see what that does? My expectation is that it will corrupt the stack.

Unfortunately, I don't have experience with the CallingConventionConverter codebase.

Reply:
You were right! I always got back a `0` when the argument type was a `float`, regardless of its value.